### PR TITLE
[NEW][APPS] Allow apps to open contextual bar

### PR DESCRIPTION
**Pull request by thassiov**
## Proposed changes (including videos or screenshots)

Opens a contextual bar using app ui interactions (`CONTEXTUAL_BAR_OPEN`)

https://user-images.githubusercontent.com/733282/146704076-d2d115f2-6ca6-4ed0-b450-81be580889a4.mp4

## Steps to test or reproduce

In this example we are using the _poll app_ and getting the `poll` slashcommand's contents to populate the contextual bar.

## Further comments

Depends on https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/447

### CODE REVIEW

File: app/ui-message/client/ActionManager.js [hunk 1]
```diff
@@ -172,6 +193,20 @@ export const triggerCancel = async ({ view, ...options }) => {
 	}
 };
 
+export const getUserInteractionPayloadByViewId = (viewId) => {
+	if (!viewId) {
+		return {};
```
Comment (d-gubert) [inline app/ui-message/client/ActionManager.js:198]: Wondering if we shouldn't throw an error here 🤔

File: client/views/room/Header/ToolBox/ToolBox.tsx [hunk 2]
```diff
@@ -36,29 +37,42 @@ const ToolBox: FC<ToolBoxProps> = ({ className }) => {
...
-				[item.id]: item.renderOption || renderMenuOption,
-			};
-			return [
-				item.id,
-				{
-					label: { title: t(item.title), icon: item.icon },
-					action: (): void => {
-						openTabBar(item.id);
+		(isMobile ? actions : actions.slice(6))
+			.filter((action) => !action.unlisted)
```
Comment (d-gubert) [inline client/views/room/Header/ToolBox/ToolBox.tsx:41]:
```suggestion
			.filter((action) => !action.unlisted)
```

File: client/views/room/Header/ToolBox/ToolBox.tsx [hunk 3]
```diff
@@ -36,29 +37,42 @@ const ToolBox: FC<ToolBoxProps> = ({ className }) => {
...
+		const openAppsContextualBar = (): void => {
+			openTabBar('apps-contextual-bar');
+		};
+
+		on('open-apps-contextual-bar', openAppsContextualBar);
+		return (): void => {
+			off('open-apps-contextual-bar', openAppsContextualBar);
+		};
+	});
+
```
Comment (d-gubert) [inline client/views/room/Header/ToolBox/ToolBox.tsx:75]: is this effect still needed?

File: client/views/room/contextualBar/Apps/Apps.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import { UiKitComponent, UiKitModal, modalParser } from '@rocket.chat/fuselage-ui-kit';
+import React from 'react';
+
+import VerticalBar from '../../../../components/VerticalBar';
+
+const Apps = ({ view, onSubmit, onClose }): JSX.Element => (
+	<>
+		<VerticalBar.Header>
+			{view.showIcon ? <VerticalBar.Icon name='keyboard' /> : null}
+			<VerticalBar.Text>{'App bar'}</VerticalBar.Text>
```
Comment (d-gubert) [inline client/views/room/contextualBar/Apps/Apps.tsx:11]: This title should come from the `view`

File: client/views/room/contextualBar/Apps/Apps.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { ButtonGroup, Button, Box } from '@rocket.chat/fuselage';
+import { UiKitComponent, UiKitModal, modalParser } from '@rocket.chat/fuselage-ui-kit';
+import React from 'react';
+
+import VerticalBar from '../../../../components/VerticalBar';
+
+const Apps = ({ view, onSubmit, onClose }): JSX.Element => (
+	<>
+		<VerticalBar.Header>
+			{view.showIcon ? <VerticalBar.Icon name='keyboard' /> : null}
```
Comment (d-gubert) [inline client/views/room/contextualBar/Apps/Apps.tsx:10]: The icon should be the app's own icon

File: client/views/room/lib/Toolbox/defaultActions.ts [hunk 6]
```diff
@@ -76,3 +76,13 @@ addAction('keyboard-shortcut-list', {
...
+
+addAction('apps-contextual-bar', {
+	groups: ['channel', 'group', 'direct', 'direct_multiple', 'team'],
+	id: 'apps-contextual-bar',
+	title: 'Apps Contextual Bar',
+	icon: 'members',
+	template: lazy(() => import('../../contextualBar/Apps')),
+	order: 300,
+	unlisted: true,
+});
```
Comment (d-gubert) [inline client/views/room/lib/Toolbox/defaultActions.ts:88]:
```suggestion
```

File: client/views/room/lib/Toolbox/index.tsx [hunk 7]
```diff
@@ -32,6 +32,7 @@ export type ToolboxActionConfig = {
 	hotkey?: string;
 	action?: (e: MouseEvent<HTMLElement>) => void;
 	template?: string | FC | LazyExoticComponent<FC<{ rid: string; tabBar: any }>>;
+	unlisted?: boolean;
```
Comment (d-gubert) [inline client/views/room/lib/Toolbox/index.tsx:35]:
```suggestion
```

File: client/views/room/Header/ToolBox/ToolBox.tsx (hunk 3 above)
Comment (thassiov) [reply client/views/room/Header/ToolBox/ToolBox.tsx]: It was removed

File: client/views/room/contextualBar/Apps/Apps.tsx (hunk 5 above)
Comment (thassiov) [reply client/views/room/contextualBar/Apps/Apps.tsx]: It has the icon and the app's name :+1:

File: client/views/room/contextualBar/Apps/Apps.tsx (hunk 4 above)
Comment (thassiov) [reply client/views/room/contextualBar/Apps/Apps.tsx]: It is comming from an object with the app's name and id

**Review by d-gubert — APPROVED** [review]

**Pull request merged**